src/world/action/ultra_boots.c

```diff
@@ -1,5 +1,200 @@
+                playerStatus->spriteFacingAngle += 360.0f;
+            }
+        }
+    }
+    if (playerStatus->gravityIntegrator[0] <= 0.0f) {
+        *(&gSpinHistoryPosY + (gSpinHistoryBufferPos)) = playerStatus->position.y;
+    }
+
+    *(&gSpinHistoryPosAngle + gSpinHistoryBufferPos++) = playerStatus->spriteFacingAngle;
+
+    if (gSpinHistoryBufferPos >= 6) {
+        gSpinHistoryBufferPos = 0;
+    }
+
+    switch (playerStatus->fallState) {
+    case 0:
+        temp_f20 = integrate_gravity();
+        playerStatus->position.y = player_check_collision_below(temp_f20, &sp10);
+        if (sp10 >= 0 && collisionStatus->currentFloor & 0x4000 && (get_entity_type(collisionStatus->currentFloor) - 7 < 2)) {
+            currentFloorEntity = get_entity_by_index(collisionStatus->currentFloor);
+            currentFloorEntity->collisionFlags |= 1;
+            disable_player_input();
+            playerStatus->fallState = 0xB;
+        } else {
+            if (temp_f20 <= 0.0f) {
+                record_jump_apex();
+                playerStatus->framesOnGround = 3;
+                playerStatus->flags |= 4;
+                playerStatus->fallState++;
+                sfx_play_sound_at_player(0x147, 0);
+            }
+            if (sp10 >= 0) {
+                playerStatus->flags &= ~0x00020008;
+                set_action_state(ACTION_STATE_LAND);
+            }
+        }
+        break;
+    case 1:
+        if (--playerStatus->framesOnGround <= 0) {
+            playerStatus->fallState++;
+        }
+        break;
+    case 2:
+        temp_f20 = integrate_gravity();
+        playerStatus->position.y = player_check_collision_below(temp_f20, &sp10);
+        if (temp_f20 < -100.0f) {
+            playerStatus->gravityIntegrator[3] = 0.0f;
+            playerStatus->gravityIntegrator[2] = 0.0f;
+            playerStatus->gravityIntegrator[1] = 0.0f;
+            playerStatus->gravityIntegrator[0] = -100.0f;
+        }
+        if (sp10 >= 0) {
+            if (collisionStatus->currentFloor & 0x4000) {
+                temp_v0 = get_entity_type(collisionStatus->currentFloor);
+                if (temp_v0 - 0x2E < 2) {
+                    playerStatus->flags &= ~0x00020008;
+                    set_action_state(ACTION_STATE_LAND);
+                    return;
+                }
+                if (temp_v0 - 7 < 2) {
+                    playerStatus->flags &= ~0x00020008;
+                    phys_player_land();
+                    exec_ShakeCam1(0, 0, 4);
+                    sfx_play_sound_at_player(0x14A, 0);
+                    start_rumble(0x100, 0x32);
+
+                    tempStruct = &D_8015A578;
+                    tempStruct->unk_00 = 1;
+                    tempStruct->unk_08 = playerStatus->position.y;
+                    playerStatus->flags |= 0x400;
+                    return;
+                }
+            }
+            temp_v1_2 = get_collider_type_by_id(sp10) & 0xFF;
```

Comment:
Change this variable's type to `s8` to get rid of this explicit cast

Reply:
that'd be u8